### PR TITLE
Switch out unicorn for puma

### LINKED ISSUE #112: Migrate the app to puma
Puma is kinda becoming the de facto web server to serve Rails apps nowadays as it's become the default server for new Rails apps. This app shouldn't be doing anything that isn't thread-safe, so it should be no real issue to switch.

### PR DESCRIPTION
**Pull request by t27duck**
Closes #112 

The puma web server is more or less the de-facto server for Rails. It can also be used for local development instead of Webrick.

Replaced the gems, updated the Procfile to match Heroku's docs, and implemented the default puma configuration with some minor tweaks.

Specs (including capybara tests) pass. Local app comes up no issue as well using puma as a local dev server.

### CODE REVIEW

File: config/puma.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+# Specifies the `pidfile` that Puma will use.
+pidfile ENV.fetch("PIDFILE") { "tmp/pids/server.pid" }
+
+# Specifies the number of `workers` to boot in clustered mode.
+# Workers are forked web server processes. If using threads and workers together
+# the concurrency of the application would be max `threads` * `workers`.
+# Workers do not work on JRuby or Windows (both of which do not support
+# processes).
+#
+workers ENV.fetch("WEB_CONCURRENCY") { 3 }
```
Comment (t27duck) [inline config/puma.rb:28]: This is default off with a fallback value of 2. This matches the old unicorn config.

File: config/puma.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+# Use the `preload_app!` method when specifying a `workers` number.
+# This directive tells Puma to first boot the application and load code
+# before forking the application. This takes advantage of Copy On Write
+# process behavior so workers use less memory.
+#
+preload_app!
```
Comment (t27duck) [inline config/puma.rb:35]: This matches the old unicorn config, may as well.

**Comment by t27duck** [comment]
@armilam - Not sure if Miles wanted to look at this since he responded originally to the issue, but this should contain all the steps I talked about in my reply to him.

**Comment by armilam** [comment]
@t27duck thank you!

**Pull request merged**